sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs

```diff
@@ -239,5 +246,855 @@ public virtual Response<TimeSeriesModelSettings> UpdateModelSettingsDefaultTypeI
+                    scope.Start();
+
+                    try
+                    {
+                        Response<GetInstancesPage> getInstancesResponse = _timeSeriesInstancesRestClient.List(null, _clientSessionId, cancellationToken);
+                        return Page.FromValues(getInstancesResponse.Value.Instances, getInstancesResponse.Value.ContinuationToken, getInstancesResponse.GetRawResponse());
+                    }
+                    catch (Exception ex)
+                    {
+                        scope.Failed(ex);
+                        throw;
+                    }
+                }
+
+                Page<TimeSeriesInstance> NextPageFunc(string nextLink, int? pageSizeHint)
+                {
+                    using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(GetInstances)}");
+                    scope.Start();
+
+                    try
+                    {
+                        Response<GetInstancesPage> getInstancesResponse = _timeSeriesInstancesRestClient.List(nextLink, _clientSessionId, cancellationToken);
+                        return Page.FromValues(getInstancesResponse.Value.Instances, getInstancesResponse.Value.ContinuationToken, getInstancesResponse.GetRawResponse());
+                    }
+                    catch (Exception ex)
+                    {
+                        scope.Failed(ex);
+                        throw;
+                    }
+                }
+
+                return PageableHelpers.CreateEnumerable(FirstPageFunc, NextPageFunc);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Gets Time Series instances by instance names asynchronously.
+        /// </summary>
+        /// <param name="timeSeriesNames">List of names of the Time Series instance to return.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>
+        /// List of instance or error objects corresponding by position to the array in the request. Instance object is set when operation is successful
+        /// and error object is set when operation is unsuccessful.
+        /// </returns>
+        /// <remarks>
+        /// For more samples, see <see href="https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/samples">our repo samples</see>.
+        /// </remarks>
+        /// <example>
+        /// <code snippet="Snippet:TimeSeriesInsightsSampleGetInstancesByNames">
+        /// </code>
+        /// </example>
+        /// <exception cref="ArgumentNullException">
+        /// The exception is thrown when <paramref name="timeSeriesNames"/> is <c>null</c>.
+        /// </exception>
+        /// <exception cref="ArgumentException">
```

Comment:
Is this required?
Since we already have code at 390

Reply:
In this case yes, because they are different. ArgumentNullException is thrown when an object is null. ArgumentException is thrown when one of the parameters is invalid, in this case Empty.
So Argument.AssertNotNullOrEmpty can throw either.